core/vm/actor/builtin/types/miner/bitfield_queue.hpp

```diff
@@ -16,6 +16,10 @@ namespace fc::vm::actor::builtin::types::miner {
   using primitives::RleBitset;
 
   struct BitfieldQueue {
+    // TODO(turuslan): must work with different bits
+    //   Deadline.ExpirationsEpochs=5
+    //   Partition.EarlyTerminated=3
+    //   miner State.PreCommittedSectorsExpiry=6
```

Comment:
Actually it works, because `Amt` initialized with correct `bits` is copied into `BitfieldQueue`.
It's just that `Array<>` type with "default" `bits` may be confusing.

`bits` of `Array<bits>` are only used for default construction and cbor decoding.
Maybe can split `Array` into "value typed amt wrapper" (without bits) and subclass with "constructor and decode" (with bits).

Reply:
It is always initialized by creating. If you want to set bits directly, we can make BitfieldQueue be templated